pkg/env/certificateRefresher.go

```diff
@@ -0,0 +1,102 @@
+package env
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/keyvault"
+)
+
+type CertificateRefresher interface {
+	Start(context.Context) error
+	GetCertificates() (*rsa.PrivateKey, []*x509.Certificate)
+}
+
+type refreshingCertificate struct {
+	lock     sync.RWMutex
+	certs    []*x509.Certificate
+	key      *rsa.PrivateKey
+	interval time.Duration
+	logger   *logrus.Entry
+	kv       keyvault.Manager
+	certName string
+	stop     <-chan struct{}
+}
+
+func newCertificateRefresher(ctx context.Context, logger *logrus.Entry, interval time.Duration, kv keyvault.Manager, certificateName string, stop <-chan struct{}) CertificateRefresher {
+	r := &refreshingCertificate{
+		logger:   logger,
+		interval: interval,
+		kv:       kv,
+		certName: certificateName,
+		stop:     stop,
+	}
+
+	return r
```

Comment:
Nit: we can straight away here:

```suggestion
	return &refreshingCertificate{
		logger:   logger,
		interval: interval,
		kv:       kv,
		certName: certificateName,
		stop:     stop,
	}
```